internal/datanode/compaction/clustering_compactor.go

```diff
@@ -677,11 +677,11 @@ func (t *clusteringCompactionTask) writeToBuffer(ctx context.Context, clusterBuf
 	t.clusterBufferLocks.Lock(clusterBuffer.id)
 	defer t.clusterBufferLocks.Unlock(clusterBuffer.id)
 	// prepare
-	if clusterBuffer.writer == nil {
+	if clusterBuffer.writer.Load().(*SegmentWriter) == nil {
```

Comment:
-> clusterBuffer.writer.Load() == nil 
if the clusterBuffer.writer is nil, it will panic. Though it should not happen

Reply:
ok